.github/workflows/build.yaml

```diff
@@ -18,38 +34,86 @@ jobs:
+      with:
+        fetch-depth: 0
     - name: Set up Python
-      uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb
+      uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 # v4.1.0
       with:
         python-version: ${{ matrix.python }}
     - name: Create and activate virtual environment
-      if: ${{ matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest' }}
+      if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
       run: |
         make venv
         source .venv/bin/activate
       env:
         PYTHON: python
     - name: Create and activate virtual environment (Windows)
-      if: ${{ matrix.os == 'windows-latest' }}
+      if: matrix.os == 'windows-latest'
       run: |
         make venv
         .venv/Scripts/Activate.ps1
       env:
         PYTHON: python
     - name: Install dependencies
       run: make setup
+    - name: Create changelog and bump
+      if: github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      run: |
+        # Set up the GitHub user and email as author for the release commit.
+        git config --global user.name $USER_NAME
+        git config --global user.email $USER_EMAIL
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        cz changelog --dry-run --incremental --unreleased-version "Release Notes" > dist/RELEASE_NOTES.md
+        # Update the CHANGELOG.md.
+        cz changelog
+        cz bump
     - name: Build the package
-      # We don't need to check and test the package separately
-      # because `make dist` runs those targets first and only
-      # builds the package if they succeed.
-      run: make dist
-    - name: Build the docs
-      run: make docs
+      # We don't need to check and test the package separately because `make dist` runs
+      # those targets first and only builds the package if they succeed.
+      run: |
+        make dist requirements sbom
+    - name: Compute package hash
+      if: matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      id: compute-hash
+      shell: bash
+      run: |
+        set -euo pipefail
+        # Find the paths to the files that will be included in the release.
+        TARBALL_PATH=$(find dist -name "*.tar.gz")
+        WHEEL_PATH=$(find dist -name "*.whl")
+        SBOM_PATH=$(find dist -name "*-sbom.json")
+        HTML_DOCS_PATH=$(find dist -name *-docs-html.zip)
+        # Make sure dist/RELEASE_NOTES.md (which contains the release notes) exists.
+        touch dist/RELEASE_NOTES.md
+        NOTES_PATH=$(find dist -name RELEASE_NOTES.md)
+        # Compute the sha digest for all the release files and encode them using base64.
+        DIGEST=$(sha256sum $TARBALL_PATH $WHEEL_PATH $SBOM_PATH $HTML_DOCS_PATH $NOTES_PATH | base64 -w0)
+        echo "digest of artifacts is $DIGEST."
```

Comment:
Should the `echo` be in [sentence case](https://en.wikipedia.org/wiki/Capitalization#Sentence_case)? Same question for the `echo` in the Release Action.